docs/proposals/17-shoot-control-plane-migration-bad-case.md

```diff
@@ -0,0 +1,102 @@
+
+The mechanisms and adaptations described below have been tested as part of a PoC prior to describing them here.
+
+## Owner Election / Copying Snapshots
+
+To achieve the goals outlined above, an "owner election" (or rather, "ownership passing") mechanism is introduced to ensure that the source and destination seeds are able to successfully negotiate a single "owner" during the migration. This mechanism is based on special *owner DNS records* that uniquely identify the seed that currently hosts the shoot's control plane ("owns" the shoot).
+
+For example, for a shoot named `i500152-gcp` in project `dev` that uses an internal domain suffix `internal.dev.k8s.ondemand.com` and is scheduled on a seed with an identity `shoot--i500152--gcp2-0841c87f-8db9-4d04-a603-35570da6341f-sap-landscape-dev`, the owner DNS record is a TXT record with a domain name `owner.i500152-gcp.dev.internal.dev.k8s.ondemand.com` and a single value `shoot--i500152--gcp2-0841c87f-8db9-4d04-a603-35570da6341f-sap-landscape-dev`. The owner DNS record is created and maintained by reconciling an `owner` DNSRecord resource, if the recently introduced DNSRecords feature is enabled via the `UseDNSRecords` feature gate. 
+
+Unlike other extension resources, the `owner` DNSRecord resource is not reconciled every time the shoot is reconciled, but only when the resource is created. Therefore, the owner DNS record value (the owner ID) is updated only when the shoot is migrated to a different seed. For more information, see [Add handling of owner DNSRecord resources](https://github.com/gardener/gardener/pull/4307).
+
+The owner DNS record domain name and owner ID are passed to components that need to perform ownership checks, such as the `backup-restore` container of the `etcd-main` StatefulSet, and all extension controllers. These components then check regularly whether the actual owner ID (the value of the record) matches the passed ID. If they don't, the ownership check is considered failed, which causes the special behavior described below.
+
+**Note:** A previous revision of this document proposed using "sync objects" written to and read from the backup container of the source seed as JSON files by the `etcd-backup-restore` processes in both seeds. With the introduction of owner DNS records such sync objects are no longer needed.
+
+For the destination seed to actually become the owner, it needs to acquire the shoot's etcd data by copying the final full snapshot (and potentially also older snapshots) from the backup container of the source seed.
+
+The mechanism to copy the snapshots and pass the ownership from the source to the destination seed consists of the following steps:
+
+1. The reconciliation flow ("restore" phase) is triggered in the destination seed without first executing the migration flow in the source seed (or perhaps it was executed, but it failed, and its state is currently unknown).
```

Comment:
How is this done? By manipulating the `status` of the `Shoot`? Who is doing this and when?

Reply:
This is not yet done in the PoC, we will figure the details out during the productization phase. I think we should always attempt to execute the migration flow and if it fails, or doesn't succeed within a certain amount of time, we should fall back to the "bad case" by triggering the reconciliation flow ("restore" phase) in the destination seed anyway.